model/qti/asset/handler/SharedStimulusAssetHandler.php

```diff
@@ -207,4 +208,17 @@ public function finalize()
     {
         // Nothing to do
     }
+
+    private function getSharedStimulusFactory(): SharedStimulusFactory
+    {
+        return $this->getServiceLocator()->get(SharedStimulusFactory::class);
+    }
+
+    private function buildLabelBaseOnParentPath()
+    {
+        $parentPath = $this->getParentPath();
+        $decodedParentPath = json_decode($parentPath);
```

Comment:
are you sure it is json in all cases ?

Reply:
I will check this.

Reply:
I found only one usage of this attribute in `\oat\taoQtiItem\model\qti\ImportService::importQtiItem` 
For some reason we encode and set this attribute as json. I adjust SharedStimulusHandler to existing code.